src/Razor/src/Microsoft.CodeAnalysis.Razor.CohostingShared/HtmlDocumentServices/HtmlDocumentSynchronizer.SynchronizationRequest.cs

```diff
@@ -21,16 +21,17 @@ private class SynchronizationRequest(RazorDocumentVersion requestedVersion) : ID
 
         public RazorDocumentVersion RequestedVersion => _requestedVersion;
 
-        internal static SynchronizationRequest CreateAndStart(TextDocument document, RazorDocumentVersion requestedVersion, Func<TextDocument, RazorDocumentVersion, CancellationToken, Task<SynchronizationResult>> syncFunction)
+        internal static SynchronizationRequest CreateAndStart(TextDocument document, RazorDocumentVersion requestedVersion, Func<TextDocument, RazorDocumentVersion, CancellationToken, Task<SynchronizationResult>> syncFunction, CancellationToken cancellationToken)
         {
             var request = new SynchronizationRequest(requestedVersion);
-            request.Start(document, syncFunction);
+            request.Start(document, syncFunction, cancellationToken);
             return request;
         }
 
-        private void Start(TextDocument document, Func<TextDocument, RazorDocumentVersion, CancellationToken, Task<SynchronizationResult>> syncFunction)
+        private void Start(TextDocument document, Func<TextDocument, RazorDocumentVersion, CancellationToken, Task<SynchronizationResult>> syncFunction, CancellationToken cancellationToken)
         {
-            _cts = new(TimeSpan.FromMinutes(1));
+            _cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            _cts.CancelAfter(TimeSpan.FromMinutes(1));
```

Comment:
One minute seems pretty long. Is there any chance that there's will be a bunch of uncancelled work hanging around for long periods of time?

Reply:
I agree it seems long, but I don't really know what is a good threshold for an OOP call. I'll lower it to 500ms at least I guess. Uncancelled work hanging around could in theory happen if the client never cancelled anything or there was requests for future document versions, but it doesn't seem a big concern to me.